timed/authentication.py

```diff
@@ -10,9 +13,18 @@
 from mozilla_django_oidc.auth import LOGGER, OIDCAuthenticationBackend
 from rest_framework.exceptions import AuthenticationFailed
 
+if TYPE_CHECKING:
+    from typing import Callable, Self
+
+    from django.db.models import QuerySet
+
+    from timed.employment.models import User
+
 
 class TimedOIDCAuthenticationBackend(OIDCAuthenticationBackend):
-    def get_introspection(self, access_token, _id_token, _payload):
+    def get_introspection(
+        self, access_token: str, _id_token: str, _payload: dict
```

Comment:
I take it you're not a fan of `Self`?
```suggestion
        self: Self, access_token: str, _id_token: str, _payload: dict
```

Reply:
Its mostly just not necessary, most IDEs respectively LSPs know how to handle `self` without an annotation

Reply:
Really not neccessary, as that's given by definition. `Self` can be used as return value or when another parameter of the same type is taken somewhere

Reply:
the docs call it "harmless" and i agree with y'all